src/detect.c

```diff
@@ -1366,24 +1444,10 @@ static void DetectRunTx(ThreadVars *tv,
         }
 
         /* merge 'state' rules from the regular prefilter */
+#ifdef PROFILING
         uint32_t x = array_idx;
-        for (uint32_t i = 0; i < det_ctx->match_array_cnt; i++) {
-            const Signature *s = det_ctx->match_array[i];
-            if (s->app_inspect != NULL) {
-                const SigIntId id = s->num;
-                det_ctx->tx_candidates[array_idx].s = s;
-                det_ctx->tx_candidates[array_idx].id = id;
-                det_ctx->tx_candidates[array_idx].flags = NULL;
-                det_ctx->tx_candidates[array_idx].stream_reset = 0;
-                array_idx++;
-
-                SCLogDebug("%p/%"PRIu64" rule %u (%u) added from 'match' list",
-                        tx.tx_ptr, tx.tx_id, s->id, id);
-            }
-        }
-        do_sort = (array_idx > x); // sort if match added anything
```

Comment:
Why do we need to sort these ?

Asking because below there is 
`do_sort |= (old && old != array_idx);`

which means the sorting only takes place if there were already match candidates and new ones were added from stored. 
But stored rules do not look ordered to me on every case :
If I have 4 rules
Rule 1 and 3 get stored on first packet (because partial match on first buffers, need to be confirmed, like http request line, waiting for http body)
Rule 2 and 4 get stored on first packet (because partial match on their first buffers, need to be confirmed, like http header, waiting for http body)

Then stored will have 1,3,2,4 on third packet with http body, and do_sort will be false because old==0

Thinking about this in the context of https://github.com/OISF/suricata/pull/10231

Reply:
> Why do we need to sort these ?

How can I make a SV test fail if they are not sorted in the end ?